deployment/8_upgrade_v1_2.js

```diff
@@ -127,45 +127,51 @@ const deploy = async (network) => {
     for (let idx = 0; idx < versions.length; idx++) {
         const version = versions[idx];
-        const moduleNames = MODULES_TO_DISABLE.concat(MODULES_TO_ENABLE);
-        const toRemove = version.modules.filter(module => moduleNames.includes(module.name));
-        const toKeep = version.modules.filter(module => !moduleNames.includes(module.name));
-        if(idx == 0) {
-            let modules = toKeep.concat(toAdd);
-            fingerprint = utils.versionFingerprint(modules);
-            newVersion.version = semver.lt(version.version, TARGET_VERSION)? TARGET_VERSION : semver.inc(version.version, 'patch');
+        let toAdd, toRemove;
+        if (idx == 0) {
+            const moduleNamesToRemove = MODULES_TO_DISABLE.concat(MODULES_TO_ENABLE);
+            toRemove = version.modules.filter(module => moduleNamesToRemove.includes(module.name));
+            toAdd = newModuleWrappers.map((wrapper) => {
+                return {
+                    'address': wrapper.contractAddress,
+                    'name': wrapper._contract.contractName
+                };
+            });
+            const toKeep = version.modules.filter(module => !moduleNamesToRemove.includes(module.name));
+            const modulesInNewVersion = toKeep.concat(toAdd);
+            fingerprint = utils.versionFingerprint(modulesInNewVersion);
+            newVersion.version = semver.lt(version.version, TARGET_VERSION) ? TARGET_VERSION : semver.inc(version.version, 'patch');
             newVersion.createdAt = Math.floor((new Date()).getTime() / 1000);
-            newVersion.modules = modules;
+            newVersion.modules = modulesInNewVersion;
             newVersion.fingerprint = fingerprint;
 
             ////////////////////////////////////
             // Deregister old modules
             ////////////////////////////////////
-
             for (let i = 0; i < toRemove.length; i++) {
                 await multisigExecutor.executeCall(ModuleRegistryWrapper, "deregisterModule", [toRemove[i].address]);
             }
+        } else {
+            // add all modules present in newVersion that are not present in version
+            toAdd = newVersion.modules.filter(module => !version.modules.map(m => m.address).includes(module.address));
+            // remove all modules from version that are no longer present in newVersion
+            toRemove = version.modules.filter(module => !newVersion.modules.map(m => m.address).includes(module.address));
         }
-        
+
         const UpgraderWrapper = await deployer.deploy(
             Upgrader,
             {},
-            toRemove.map((module) => {return module.address;}),
-            toAdd.map((module) => {return module.address;})
+            config.contracts.ModuleRegistry,
+            toRemove.map(module => module.address),
+            toAdd.map(module => module.address)
         );
-        const upgraderName = version.fingerprint + '_' + fingerprint; 
```

Comment:
Same as above. We should register the upgrader module as a module and a register.